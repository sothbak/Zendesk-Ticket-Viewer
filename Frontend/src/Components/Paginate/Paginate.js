import React, { Component } from 'react';

import styles from './Paginate.module.css';

class Paginate extends Component {
    handleChangePage = (page) => {
        this.props.changePage(page);
    }
    handleChangePage = this.handleChangePage.bind(this);

    render() {
        let numPages = [...Array(this.props.total_pages).keys()];

        // if there are no pages, don't render anything
        let paginateButtons = numPages.length === 0 ? null : numPages.map(number => {
            return (
                <button key={number + 1} className={styles.button} onClick={() => this.handleChangePage(number + 1)} disabled={this.props.current_page === number + 1}>{number + 1}</button>
            )
        });
        let information = numPages.length === 0 ? null : (
            <div>
                <p>{this.props.total_tickets} total ticket(s), this page is showing {this.props.page_tickets} ticket(s) </p>
                <p>Currently on page {this.props.current_page} of {this.props.total_pages} </p>
            </div>
        );

        return (
            <div>
                {information}
                {paginateButtons}
            </div>
        )
    }
}

export default Paginate;