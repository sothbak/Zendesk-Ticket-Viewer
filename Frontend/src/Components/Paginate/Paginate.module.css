.button { 
    cursor: pointer;
    color: blue;
}