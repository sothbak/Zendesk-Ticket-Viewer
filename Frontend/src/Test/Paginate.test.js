import React from 'react';
import { shallow } from 'enzyme';
import Paginate from '../Components/Paginate/Paginate.js';

describe('Paginate', () => {
  it('should render correctly', () => {
    const paginate = shallow(<Paginate total_tickets={140}
                                        page_tickets={25}
                                        current_page={1}
                                        total_pages={6} />);
    expect(paginate).toMatchSnapshot();
  });

  it('should say currently showing 25 tickets and on page 1 of 6', () => {
    const paginate = shallow(<Paginate total_tickets={140}
                                        page_tickets={25}
                                        current_page={1}
                                        total_pages={6} />);
    expect(paginate.text()).toContain("140 total ticket(s), this page is showing 25 ticket(s) Currently on page 1 of 6");
  });

  it('should say currently showing 15 tickets and on page 6 of 6', () => {
    const paginate = shallow(<Paginate total_tickets={140}
                                        page_tickets={15}
                                        current_page={6}
                                        total_pages={6} />);
    expect(paginate.text()).toContain("140 total ticket(s), this page is showing 15 ticket(s) Currently on page 6 of 6");
  });
});