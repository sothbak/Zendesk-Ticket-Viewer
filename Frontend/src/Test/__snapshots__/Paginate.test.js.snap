// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate should render correctly 1`] = `
<div>
  <div>
    <p>
      140
       total ticket(s), this page is showing 
      25
       ticket(s) 
    </p>
    <p>
      Currently on page 
      1
       of 
      6
       
    </p>
  </div>
  <button
    className="button"
    disabled={true}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="button"
    disabled={false}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="button"
    disabled={false}
    key="3"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="button"
    disabled={false}
    key="4"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="button"
    disabled={false}
    key="5"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="button"
    disabled={false}
    key="6"
    onClick={[Function]}
  >
    6
  </button>
</div>
`;
